samples/ClientSample/HubSample.cs

```diff
@@ -48,12 +48,13 @@ internal static void Register(CommandLineApplication app)
                 // Set up handler
                 connection.On<string>("Send", Console.WriteLine);
 
-                connection.Closed += e =>
+                _ = connection.Closed.ContinueWith((task, state) =>
```

Comment:
Update the sample to use a completely different pattern. Instead of using the cancellation token you can now us the task.

```C#
while (!connection.Closed.IsCompleted)
{
     var completedTask = await Task.WhenAny(Task.Run(() => Console.ReadLine()), connection.Closed);
     if (completedTask == connection.Closed)
     {
         break;
     }

     var line = await (Task<string>)completedTask;

     if (line == null)
     {
         break;
     }

     await connection.InvokeAsync<object>("Send", line);
}
```

Reply:
We should still use the Cts though to watch for the CancelKeyPress though, right?

Reply:
Hmm, maybe we can call `DisposeAsync` in CancelKeyPress

Reply:
The you would not dispose if the connection was closed in a different manner (e.g. by the server or due to a network error)

Reply:
I don't see why not.